qiskit_optimization/translators/docplex_mp.py

```diff
@@ -0,0 +1,286 @@
+            mdl.add_constraint(linear_expr >= rhs, ctname=name)
+        elif sense == Constraint.Sense.LE:
+            mdl.add_constraint(linear_expr <= rhs, ctname=name)
+        else:
+            # should never happen
+            raise QiskitOptimizationError("Unsupported constraint sense: {}".format(sense))
+
+    # add quadratic constraints
+    for i, q_constraint in enumerate(quadratic_program.quadratic_constraints):
+        name = q_constraint.name
+        rhs = q_constraint.rhs
+        if (
+            rhs == 0
+            and q_constraint.linear.coefficients.nnz == 0
+            and q_constraint.quadratic.coefficients.nnz == 0
+        ):
+            continue
+        quadratic_expr = 0
+        for j, v in q_constraint.linear.to_dict().items():
+            quadratic_expr += v * var[cast(int, j)]
+        for (j, k), v in q_constraint.quadratic.to_dict().items():
+            quadratic_expr += v * var[cast(int, j)] * var[cast(int, k)]
+        sense = q_constraint.sense
+        if sense == Constraint.Sense.EQ:
+            mdl.add_constraint(quadratic_expr == rhs, ctname=name)
+        elif sense == Constraint.Sense.GE:
+            mdl.add_constraint(quadratic_expr >= rhs, ctname=name)
+        elif sense == Constraint.Sense.LE:
+            mdl.add_constraint(quadratic_expr <= rhs, ctname=name)
+        else:
+            # should never happen
+            raise QiskitOptimizationError("Unsupported constraint sense: {}".format(sense))
+
+    return mdl
+
+
+def from_docplex_mp(model: Model) -> "QuadraticProgram":
+    """Translate a docplex.mp model into a quadratic program.
+
+    Note that this supports only basic functions of docplex as follows:
+    - quadratic objective function
+    - linear / quadratic constraints
+    - binary / integer / continuous variables
+
+    Args:
+        model: The docplex.mp model to be loaded.
+
+    Returns:
+        The quadratic program corresponding to the model.
+
+    Raises:
+        QiskitOptimizationError: if the model contains unsupported elements.
+    """
+    if not isinstance(model, Model):
+        raise QiskitOptimizationError(f"The model is not compatible: {model}")
+
+    # pylint: disable=cyclic-import
+    from qiskit_optimization.problems.quadratic_program import QuadraticProgram
+
+    quadratic_program = QuadraticProgram()
+
+    # get name
+    quadratic_program.name = model.name
+
+    # get variables
+    # keep track of names separately, since docplex allows to have None names.
+    var_names = {}
+    for x in model.iter_variables():
+        if isinstance(x.vartype, ContinuousVarType):
+            x_new = quadratic_program.continuous_var(x.lb, x.ub, x.name)
+        elif isinstance(x.vartype, BinaryVarType):
+            x_new = quadratic_program.binary_var(x.name)
+        elif isinstance(x.vartype, IntegerVarType):
+            x_new = quadratic_program.integer_var(x.lb, x.ub, x.name)
+        else:
+            raise QiskitOptimizationError(
```

Comment:
This on the other hand seems fine - we can potentially expect this to occur since the docplex model has a range of capability that is not supported by QP. In this case the user should adjust their model to remove such things. 

Question though - is there ever a case where we would want to say log a warning and continue, just taking from the model the bits we do understand and ignoring the others. Maybe its meaningless - if it made sense then perhaps an `ignore_unsupported=False` param on the method, allowing the behavior to be changed between failing hard like it does now or continuing albeit that aspects get dropped.

Reply:
If we ignore unsupported types of elements (variables, constraints, etc.) in `from_docplex_mp`, the resulting QP model will be different and the solution will be also different from what users expect. So, I would like to keep the current behavior to raise an error with unsupported type for `from_docplex_mp`.